In [ ]:
import math

A solution to problem 7 of Project Euler.

Find the 10001st prime. I build any program and see if it works before thinking about efficiency.

In [27]:
def generatePrimeList(k):
    primes = []
    t=2
    while len(primes)<k:
        if isPrime(t,primes):
            primes.append(t)
        t=t+1
    return primes

In [22]:
def isPrime(t,primes):
    isPrime = True
    for prime in primes:
        if t/prime == int(t/prime):
            isPrime = False
    return isPrime

In [28]:
primes = generatePrimeList(10001)
bigPrime = primes[-1]
print(bigPrime)

104743


The answer is 104,743, and I did not need to improve efficiency to get the answer in <1min of runtime. Clearly I can improve my program trivially by hardcoding the prime 2 in, and having the candidate <$>t<\$> increment by 2 each step. I can do even better by implementing any "proper" primality test - note I don't actually need worry too much about the slowness of checking any pseudoprimes, since most numbers are not pseudoprimes. 